### ISSUE TITLE
Integration test coverage for registry add-on

### ISSUE THREAD

**Issue by amitkrout** [minishift/minishift#1318]
This is a followup integration test coverage issue on pr #1243. Reference for creating the test scenario [#704 (comment)](https://github.com/minishift/minishift/pull/704#issuecomment-293568580)

**Comment by agajdosi** [comment]
Depending on the size of this test we could also think of covering addons in general as part of this issue - focusing on all aspects of addons command + focusing on individual addons.

**Comment by agajdosi** [comment]
Based on our conversation I am going to create a separate issue for adding a specialized test for addon command. It seems that with 4 default addons one feature file would get quite long and thus it will be better to not merge them together with broader addons command test. Also it would give us a good level of separation and better understanding of these feature files.

**Comment by amitkrout** [comment]
For exposing docker registry, shell need to be configured as per the instructed output of ```minishift docker-env```. This scenario is implemented in the ongoing pr #1453. Some part of the code can be reused for this issue once the pr #1453 merge is done.

**Comment by praveenkumar** [comment]
@amitkrout @agajdosi Does this still valid?

**Comment by amitkrout** [comment]
@praveenkumar yes, will take this on next sprint.

**Comment by amitkrout** [comment]
I have added a working registry-route feature file. I have commented some steps there which yet to be implemented/factored. To make this work i used the step ```And user waits "40" seconds``` (Not recommended). Later ```wait``` step will be removed once container status is implemented.
